bc/subscription/views.py

```diff
@@ -80,10 +84,45 @@ def post(self, request, *args, **kwargs):
         docket["case_name"] = cd["docket_name"]
         docket["case_summary"] = cd["case_summary"]
         docket["article_url"] = cd["article_url"]
-        subscription, created = create_or_update_subscription_from_docket(
-            docket
-        )
-        channels = request.POST.getlist("channels")
+        try:
+            with transaction.atomic():  # Inner atomic block, create a savepoint
+                (
+                    subscription,
+                    created,
+                ) = create_or_update_subscription_from_docket(docket)
+                channels = request.POST.getlist("channels")
+
+                # Verify that all templates produce valid post content
+                for channel_id in channels:
+                    channel = Channel.objects.get(pk=channel_id)
+                    template = get_new_case_template(
+                        channel.service, subscription.article_url
+                    )
+
+                    template.format(
+                        docket=subscription.name_with_summary,
+                        docket_link=subscription.cl_url,
+                        docket_id=subscription.cl_docket_id,
+                        article_url=subscription.article_url,
+                    )
+
+                    if not template.is_valid:
+                        raise InvalidTemplate
+        except InvalidTemplate:
+            context = {
+                "docket_id": docket_id,
+                "form": form,
+                "channels": get_channel_groups_per_user(request.user.pk),
+                "error": (
+                    "The combination of name, summary and article URL exceeds "
+                    f"the maximum character limit for {channel.get_service_display()} "
+                    "posts. Please try reducing the number of characters in the inputs."
+                    "You can use abbreviations, or remove unnecessary words. Once you "
+                    "have made these changes, resubmit the form."
+                ),
+            }
+            template = "./includes/search_htmx/case-form.html"
```

Comment:
Were there changes to this that didn't get submitted?

Reply:
I attempted several error messages that were not included in the PR, but I didn't change the case-form template, is there something strange or unusual in the code? Why are you asking? 😱

Reply:
You're right! the typo is coming from the case-form.html. I'll fix it.